src/kv/apply_changes.h

```diff
@@ -113,6 +119,11 @@ namespace kv
       // Get the version number to be used for this commit.
       version = f();
 
+      if (!new_maps.empty() && version > 0)
```

Comment:
Please add a comment explaining this. Is it "If this creates any maps, it must be fully serialised"?